parser.go

```diff
@@ -792,32 +769,21 @@ func (p *parser) parseSelector() (result Selector, err error) {
 			return
 		}
 
-		c, err := p.parseSimpleSelectorSequence()
+		c, err = p.parseSimpleSelectorSequence()
 		if err != nil {
-			return nil, err
-		}
-
-		switch combinator {
-		case ' ':
-			result = descendantSelector(result, c)
-		case '>':
-			result = childSelector(result, c)
-		case '+':
-			result = siblingSelector(result, c, true)
-		case '~':
-			result = siblingSelector(result, c, false)
+			return
 		}
+		result = combinedSelector{first: result, combinator: combinator, second: c}
 	}
-
-	panic("unreachable")
 }
 
 // parseSelectorGroup parses a group of selectors, separated by commas.
-func (p *parser) parseSelectorGroup() (result Selector, err error) {
-	result, err = p.parseSelector()
+func (p *parser) parseSelectorGroup() (result selectorGroup, err error) {
```

Comment:
The result type should be Matcher, so that in the simple case of a single selector, you can return it directly instead of wrapping it.